packages/next/src/client/components/globals/intercept-console-error.ts

```diff
@@ -10,8 +117,22 @@ export function patchConsoleError() {
 
   const originConsoleError = window.console.error
   window.console.error = (...args) => {
-    // See https://github.com/facebook/react/blob/d50323eb845c5fde0d720cae888bf35dedd05506/packages/react-reconciler/src/ReactFiberErrorLogger.js#L78
-    const error = process.env.NODE_ENV !== 'production' ? args[1] : args[0]
+    let error
+    if (
+      process.env.NODE_ENV !== 'production' &&
+      args[0] === '%o\n\n%s\n%s\n\n%s' &&
+      args[1] instanceof Error
+    ) {
+      // This looks like React printing an error object. We'll just take the error object.
+      // See https://github.com/facebook/react/blob/d50323eb845c5fde0d720cae888bf35dedd05506/packages/react-reconciler/src/ReactFiberErrorLogger.js#L78
+      error = args[1]
+    } else if (args.length === 1 && args[0] instanceof Error) {
+      error = args[0]
+    } else if (process.env.NODE_ENV !== 'production') {
+      // Format the console arguments as an error object.
+      // Keep the stack as the stack of the console.error call.
+      error = new Error(formatConsoleMessage(args))
```

Comment:
Just to confirm, since `handleClientError` is only made aware of error handlers after hydration (they're initialized inside of a `useEffect`), this means `console.error` inside the SSR render won't show the error overlay. It'd just be errors after hydration. Is that expected?

Reply:
hm. No that's not expected. Because we don't replay errors that happened on the server during SSR - we expect them to typically error again on the client and that's the one that shows.

Reply:
I think it also means it's a bit racey in the post-hydration case as well as to whether or not it'll show up, so we might not have the error handlers ready yet unless we collect the errors and show them once available. e.g. this won't trigger the overlay either (I believe since `useEffect` is invoked bottom-up?):

```tsx
"use client"

export default function Page() {
  useEffect(() => {
    console.error(new Error('kaboom'))
  }, [])
  return <h1>Home</h1>
}
```

Reply:
Feel free to take over the PR and do it properly. Mainly wanted to show case turning formatting into a string.

Reply:
Yea I'm surprised that works because hydration errors are also logged before useEffect.

I think we need to queue them.

Reply:
Another issue is that it seems to stop queuing them after that.